File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    s = set()
    n = len(elements)

    for length in range(1, n + 1):
        current_sum = sum(elements[:length])
        s.add(current_sum)

        for i in range(1, n):
            current_sum += elements[(i + length - 1) % n]
            current_sum -= elements[(i - 1) % n]
            s.add(current_sum)

    return len(s)

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    const set = new Set();
    
    for (let l = 1; l <= elements.length; l++) {
        for (let i = 0; i < elements.length; i++) {
            let sum = 0;
            for (let j = i; j < i + l; j++) {
                sum += elements[j % elements.length];
            }
            set.add(sum);
        }
    }
    
    return set.size;
}
